app/components/NavBar/components/UserSettingsDialog/index.tsx

```diff
@@ -0,0 +1,105 @@
+import React, {
+  useState, useEffect, useContext, useCallback,
+} from 'react';
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  Button,
+  Checkbox,
+  FormControlLabel,
+} from '@mui/material';
+
+import api from '@/services/api';
+import AsyncButton from '@/components/AsyncButton';
+
+import snackbar from '@/services/SnackbarUtils';
+import ConfirmContext from '@/context/ConfirmContext';
+import useConfirmDialog from '@/hooks/useConfirmDialog';
+import { UserType } from '@/common';
+
+import './index.scss';
+
+type UserSettingsDialogProps = {
+  editData: Partial<UserType>;
+  isOpen: boolean;
+  onClose: (newData?: Partial<UserType>) => void;
+  showErrorSnackbar: (message: string) => void;
+};
+
+const UserSettingsDialog = ({
+  editData,
+  isOpen = false,
+  onClose,
+  showErrorSnackbar,
+}: UserSettingsDialogProps): JSX.Element => {
+  const { showConfirmDialog } = useConfirmDialog();
+  const { isSigned } = useContext(ConfirmContext);
+  const [checkboxSelected, setCheckboxSelected] = useState(false);
+  const [isApiCalling, setIsApiCalling] = useState(false);
+
+  useEffect(() => {
+    if (editData) {
+      setCheckboxSelected(editData.allowNotifications);
+    }
+  }, [editData]);
+
+  const handleCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCheckboxSelected(event.target.checked);
+  };
+
+  const handleSubmit = useCallback(async () => {
+    if (checkboxSelected !== editData.allowNotifications) {
+      setIsApiCalling(true);
+      const req = api.put(
+        `/user/${editData.ident}`,
+        { allowNotifications: checkboxSelected },
+        {},
+      );
+      try {
+        if (isSigned) {
+          showConfirmDialog(req);
+          setIsApiCalling(false);
+        } else {
+          await req.request();
+          onClose({ ...editData, allowNotifications: checkboxSelected });
+          snackbar.success('User Settings updated successfully.');
+        }
+      } catch (err) {
+        showErrorSnackbar(`Error updating user settings: ${err.message}`);
+        onClose();
+      } finally {
+        setIsApiCalling(false);
+      }
+    } else {
+      onClose();
+    }
+  }, [checkboxSelected, editData, isSigned, showConfirmDialog, onClose, showErrorSnackbar]);
+
+  return (
+    <Dialog open={isOpen} maxWidth="sm" fullWidth className="edit-dialog">
+      <DialogTitle>Edit User Settings</DialogTitle>
+      <DialogContent>
+        <div>
+          <FormControlLabel
+            label="Allow email notifications?"
```

Comment:
can we change this to 'Allow email notifications to be sent to <user's email>', with no question mark (specifiying the email in case users wonder if the notifications they're permitting are for or about them), and put the check box on the right hand side of the text?

Reply:
Noting here that this was discussed in another forum and we decided on left hand side check box